pkg/distributor/ha_tracker.go

```diff
@@ -79,61 +82,51 @@ func (r *ReplicaDesc) mergeWithTime(mergeable memberlist.Mergeable) (memberlist.
 		return nil, nil
 	}
 
-	thisRDesc := r
-	otherRDesc := other
-
-	if otherRDesc.Replica == thisRDesc.Replica {
+	if other.Replica == r.Replica {
 		// Keeping the one with the most recent receivedAt timestamp
-		if otherRDesc.ReceivedAt > thisRDesc.ReceivedAt {
-			*thisRDesc = *other
-		} else if thisRDesc.ReceivedAt == otherRDesc.ReceivedAt && thisRDesc.DeletedAt == 0 && otherRDesc.DeletedAt != 0 {
-			*thisRDesc = *other
+		if other.ReceivedAt > r.ReceivedAt {
+			*r = *other
+		} else if r.ReceivedAt == other.ReceivedAt && r.DeletedAt == 0 && other.DeletedAt != 0 {
+			*r = *other
 		}
 	} else {
 		// keep the most recent Elected to reach consistency
-		if otherRDesc.ElectedAt > thisRDesc.ElectedAt {
-			*thisRDesc = *other
-		} else if otherRDesc.ElectedAt == thisRDesc.ElectedAt {
+		if other.ElectedAt > r.ElectedAt {
+			*r = *other
+		} else if other.ElectedAt == r.ElectedAt {
 			// if the timestamps are equal we compare receivedAt
-			if otherRDesc.ReceivedAt > thisRDesc.ReceivedAt {
-				*thisRDesc = *other
+			if other.ReceivedAt > r.ReceivedAt {
+				*r = *other
 			}
 		}
 	}
 
-	// No changes - return
-	if *thisRDesc != *otherRDesc {
+	// No changes
+	if *r != *other {
 		return nil, nil
 	}
 
 	out := NewReplicaDesc()
-	*out = *thisRDesc
+	*out = *r
 	return out, nil
 }
 
-// MergeContent noOp currently
+// MergeContent describes content of this Mergeable.
+// Given that ReplicaDesc can have only one instance at a time, it returns the ReplicaDesc it contains.
 func (r *ReplicaDesc) MergeContent() []string {
 	result := []string(nil)
 	if len(r.Replica) != 0 {
```

Comment:
WDYT about using `r.IsEmpty()` here?

Reply:
Cannot find this function as part of the ReplicaDesc, do you want me to implement it ?

Reply:
my mistake. I saw it on another protobuf type and assumed they all have this method.